src/iuse_actor.cpp

```diff
@@ -1181,6 +1181,12 @@ bool firestarter_actor::prep_firestarter_use( const player &p, tripoint &pos )
         p.add_msg_if_player( m_info, _( "There is already a fire." ) );
         return false;
     }
+    // check if there's a fire fuel source spot
+    if( warn_fuel_burn( pos ) ) {
```

Comment:
Extracting code to helpers is one thing, but it seems a bit excessive to extract this one line that we're unlikely to reuse at the cost of 8 lines of new code elsewhere.
```suggestion
    if( here.tr_at( pos ).id == tr_firewood_source ) {
```

Reply:
Oh, I agree, but apparently the firewood source is a "kludge" by making it a trap.
So my thought process was making a helper function whose guts can be exchanged later when it's unkludged, and since it's such a seldomly called check...a function call should not be that expensive/bloaty.

Aka, the tradeoff of a function that might be able to stay like this in iuse_actor, as small and unassuming as it is now, but also seldomly called (and thus not incurring the cost of a function call that often), not that 'fat' , to later just change the detection code in the file that deals with it right there.

Reply:
Expense is not an issue, but code bloat is.  This doesn't help maintainability because this is not the responsibility of the activity handlers module.

Reply:
Ah, alright. Going to change that, then.